src/main/java/com/google/firebase/database/core/Context.java

```diff
@@ -130,8 +130,12 @@ private void restartServices() {
 
   void stop() {
     stopped = true;
-    eventTarget.shutdown();
-    runLoop.shutdown();
+    if (eventTarget != null) {
+      eventTarget.shutdown();
```

Comment:
should you set them to null too?

Reply:
I think we need them to remain non-null for `Context.restartServices()` to work.